PluginClasses/ServiceApi.swift

```diff
@@ -0,0 +1,115 @@
+//
+//  ServiceApi.swift
+//  TVADataBinder
+//
+//  Created by MSApps on 16/04/2020.
+//
+
+enum HTTPStatusCode: Int{
+    case success = 200
+    case notFound = 404
+    case deleted = 204
+}
+
+import Foundation
+
+let tokenKey = "AWSCognitoTokenKey"
+let defaults = UserDefaults.standard
+var baseApi :String?
+let favouritePath = "/userlists/favorites/"
+
+/*
+Get favorite state for item according to the item identifier
+success response is for item is favorite
+not found response is for item is not favorite
+*/
+func getFavouriteState(uid: String, completion: @escaping (( _ on: Bool?) -> Void)){
+    guard let token = defaults.string(forKey: tokenKey), let baseApiUrl = baseApi else{
+        return
+    }
+    let apiName = "\(baseApiUrl)\(favouritePath)\(uid)"
+    let url = URL(string: apiName)
+    guard let apiUrl = url else{
+        return
+    }
+    let request = NSMutableURLRequest(url: apiUrl)
+    request.setValue("Bearer \(token)", forHTTPHeaderField: "Authorization")
+    request.httpMethod = "GET"
+    request.setValue("application/json", forHTTPHeaderField: "Content-Type")
+    makeRequest(request: request) { (responseJson, response, error) in
+        if(response?.statusCode == HTTPStatusCode.success.rawValue){
+            completion(true)
+        }else if (response?.statusCode == HTTPStatusCode.notFound.rawValue){
+            completion(false)
+        }else{
+            completion(nil)
+        }
+    }
+}
+
+ /*
+ Set favorite state for item according to the the item identifier
+ for set on favorite, success response is for completion(true)
+ for set off favorite, success , not found and deleted response is for completion(true)
+ */
+func setFavoriteState(uid: String, on: Bool, completion: @escaping (( _ success: Bool) -> Void)){
+    guard let token = defaults.string(forKey: tokenKey), let baseApiUrl = baseApi else{
+        return
+    }
+    let apiName = "\(baseApiUrl)\(favouritePath)\(uid)"
+    let url = URL(string: apiName)
+    guard let apiUrl = url else{
+        return
+    }
+    let request = NSMutableURLRequest(url: apiUrl)
+    request.setValue("Bearer \(token)", forHTTPHeaderField: "Authorization")
+    request.httpMethod = on ? "PUT" : "DELETE"
+    request.setValue("application/json", forHTTPHeaderField: "Content-Type")
+    if(on){
+        let params: [String:String] = ["object_uid" : uid]
+        request.httpBody = try! JSONSerialization.data(withJSONObject: params, options: .prettyPrinted)
+    }
+    makeRequest(request: request) { (responseJson, response, error) in
+        if(response?.statusCode == HTTPStatusCode.success.rawValue){
+            completion(true)
+        }else if(response?.statusCode == HTTPStatusCode.notFound.rawValue){
```

Comment:
use filter method https://stackoverflow.com/questions/30828074/check-if-array-contains-part-of-a-string-in-swift

Reply:
done